flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -852,6 +852,15 @@ protected void deregisterStreamConsumer() {
         StreamConsumerRegistrarUtil.deregisterStreamConsumers(configProps, streams);
     }
 
+    /**
+     * Gracefully stops shardConsumersExecutor without interrupting running threads. Allows test to
+     * override this to simulate exception for shutdown logic.
+     */
+    @VisibleForTesting
```

Comment:
You have marked this as `@VisibleForTesting`, but it is not called from any tests, can you make it private?